src/integrations/xiaomi-mi/xiaomi-mi.service.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  Injectable,
+  Logger,
+  OnApplicationBootstrap,
+  OnModuleInit
+} from '@nestjs/common';
+import noble, { Peripheral, Advertisement } from '@abandonware/noble';
+import { EntitiesService } from '../../entities/entities.service';
+import { ConfigService } from '../../config/config.service';
+import { XiaomiMiConfig } from './xiaomi-mi.config';
+import { makeId } from '../../util/id';
+import { SchedulerRegistry } from '@nestjs/schedule';
+import {
+  SERVICE_DATA_UUID,
+  ServiceData,
+  Parser,
+  Event,
+  EventTypes
+} from './parser';
+import { Sensor } from '../../entities/sensor';
+import { EntityCustomization } from '../../entities/entity-customization.interface';
+import { SensorConfig } from '../home-assistant/sensor-config';
+
+class SensorSet {
+  private readonly name: string;
+  private readonly entitiesService: EntitiesService;
+  private measures: { [kind: string]: Sensor } = {};
+
+  constructor(name: string, entitiesService: EntitiesService) {
+    this.name = name;
+    this.entitiesService = entitiesService;
+  }
+
+  recordMeasure(kind: string, units: string, state: number | string) {
```

Comment:
To me it looks like this class partially duplicates the entity registry functionality - imo it would be cleaner to include this method in the `XiaomiMiService`. As you have a fixed format for the ids you can simply check if the entity registry already has something registered under the id - if yes, get it from the registry with the id; if no, create a new entity as you did here.

Reply:
Done.  Thanks for the pointer!